### PR TITLE
Adding support for redis storage engine

### PR DESCRIPTION
**Pull request by halkeye**
* Convert store to be a generic interface
* Storage systems into an object
* User to object

### CODE REVIEW

File: lib/store/redis.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+	data["updated"] = user.Updated.Format("01-02-2006")
+	s.client.HMSet("user:"+user.ID, data)
+}
+
+// GetUser will load a user from redis
+func (s RedisStore) GetUser(id string) *User {
+	data, err := s.client.HGetAll("user:" + id).Result()
+	// FIXME - return err
+	if err != nil {
+		panic(err)
```
Comment (halkeye) [inline lib/store/redis.go]: the convention is returning nil, err all the way back to the top so error handling can be done in a single placement.

But I also wanted to keep to your conventions.

**Comment by halkeye** [comment]
closes #11

File: lib/store/redis.go (hunk 1 above)
Comment (XanderStrike) [reply lib/store/redis.go]: My convention is bad and I should feel bad. I respect you not wanting to go through that refactor though.

Comment (halkeye) [reply lib/store/redis.go]: Oh I will eventually. Like I said earlier I want to use this as an excuse to learn more go. Learn testing practices. Etc

I already learned a ton about interfaces.

This redis change lets me run the system locally in my k8s but not needing a filesystems.

**Comment by halkeye** [comment]
I'm not sure how the glide stuff works, so I'm not sure what needs updating and vendor and stuff

**Comment by XanderStrike** [comment]
I'll run that, it's part of the docker image building which doesn't happen automagically yet. If you've got docker installed you should be able to run `build.sh` with the first line uncommented, but I don't mind doing it for now.

**Pull request merged**